blog/2024-02-02-migrating-machines-to-xstate-v5/index.mdx

```diff
@@ -0,0 +1,565 @@
+---
+title: Migrating machines to XState v5
+description: A walkthrough of migrating an existing XState v4 machine to XState v5
+tags:
+  [
+    stately,
+    studio,
+    breaking changes,
+    migration,
+    XState v5,
+    state machines,
+    xstate,
+  ]
+authors: [kevin]
+date: 2024-02-02
+slug: 2024-02-02-migrating-machines-to-xstate-v5
+image: /blog/2024-02-02-migrating-machines-to-xstate-v5.png
+---
+
+We recently announced the recent release of XState v5! During its beta phase, we created a [migration guide](/docs/migration), specifically to call out breaking changes and to give developers onging updates regarding API changes. This post is a walkthrough of migrating existing XState machines from v4 to v5 and is intended to be more of a step-by-step companion to the migration guide. It also focuses on migrating [XState machines that are using TypeScript](/docs/typescript).
+
+{/* truncate */}
+
+:::info
+
+Read [David’s blog post on the launch of XState v5](/blog/2023-12-01-xstate-v5).
+
+Prefer video? [Watch our XState v5 webinar on YouTube](https://www.youtube.com/live/TRVjeil-y74):
+
+<YouTube id="TRVjeil-y74" />
+
+:::
+
+We at Stately have dozens of XState machines in our own [Stately Studio](https://stately.ai/studio) code base and have been migrating those to XState v5 as well. I've learned a lot about the migration process, especially after consulting my expert teammate, Mateusz (@andarist), and I wanted to share some tips from my experience to make yours even smoother.
```

Comment:
@Andarist Should I link to your GitHub or just mention you by real name only? What do you like here?